src/xlsx.rs

```diff
@@ -172,61 +186,134 @@ impl ExcelReader for Xlsx {
+                    .and_then(|o| o.ok_or_else(|| "Cell missing 'r' attribute tag".into()))
+                    .and_then(get_row_column)?;
+
                 loop {
                     match xml.next() {
                         Some(Err(e)) => return Err(e.into()),
-                        Some(Ok(Event::Start(ref e))) => match e.name() {
-                            b"v" => {
-                                // value
-                                let v = xml.read_text_unescaped(b"v")?;
-                                let value = match c_element.attributes()
-                                    .filter_map(|a| a.ok())
-                                    .find(|&(k, _)| k == b"t") {
-                                        Some((_, b"s")) => {
-                                            // shared string
-                                            let idx: usize = v.parse()?;
-                                            DataType::String(strings[idx].clone())
-                                        },
-                                        Some((_, b"str")) => {
-                                            // regular string
-                                            DataType::String(v)
-                                        },
-                                        Some((_, b"b")) => {
-                                            // boolean
-                                            DataType::Bool(v != "0")
-                                        },
-                                        Some((_, b"e")) => {
-                                            // error
-                                            DataType::Error(v.parse()?)
-                                        },
-                                        _ => v.parse().map(DataType::Float)?,
-                                    };
-                                cells.push(Cell::new(pos, value));
-                                break;
-                            },
-                            b"f" => (), // formula, ignore
-                            _name => return Err("not v or f node".into()),
-                        },
-                        Some(Ok(Event::End(ref e))) if e.name() == b"c" => break,
+                        Some(Ok(Event::Start(ref e))) => {
+                            debug!("e: {:?}", e);
+                            match e.name() {
+                                b"is" => {
+                                    cells.push(Cell::new(pos, read_inline_str(xml)?));
+                                    break;
+                                }
+                                b"v" => {
+                                    cells.push(Cell::new(pos, read_value(xml, strings, c_element)?));
+                                    break;
+                                }
+                                b"f" => {} // ignore f nodes
+                                n => return Err(format!("not a v, f, or is node: {:?}", n).into()),
```

Comment:
if possible can you add some quote (because of the not a, I find it a little more complicated to read:
not a 'v, 'f' or 'is' node